irksome/galerkin_stepper.py

```diff
@@ -0,0 +1,295 @@
+    :arg t: a :class:`Function` on the Real space over the same mesh as
+         `u0`.  This serves as a variable referring to the current time.
+    :arg dt: a :class:`Function` on the Real space over the same mesh as
+         `u0`.  This serves as a variable referring to the current time step.
+         The user may adjust this value between time steps.
+    :arg u0: A :class:`firedrake.Function` containing the current
+            state of the problem to be solved.
+    :arg bcs: An iterable of :class:`firedrake.DirichletBC` containing
+            the strongly-enforced boundary conditions.  Irksome will
+            manipulate these to obtain boundary conditions for each
+            stage of the method.
+    :arg basis_type: A string indicating if the standard (continuous)
+            Lagrange basis is to be used for the trial space or if the Bernstein
+            basis is to be used instead
+    :arg quadrature: A :class:`FIAT.QuadratureRule` indicating the quadrature
+            to be used in time, defaulting to GL with order points
+    :arg solver_parameters: A :class:`dict` of solver parameters that
+            will be used in solving the algebraic problem associated
+            with each time step.
+    :arg appctx: An optional :class:`dict` containing application context.
+            This gets included with particular things that Irksome will
+            pass into the nonlinear solver so that, say, user-defined preconditioners
+            have access to it.
+    :arg nullspace: A list of tuples of the form (index, VSB) where
+            index is an index into the function space associated with
+            `u` and VSB is a :class: `firedrake.VectorSpaceBasis`
+            instance to be passed to a
+            `firedrake.MixedVectorSpaceBasis` over the larger space
+            associated with the Runge-Kutta method
+    """
+    def __init__(self, F, order, t, dt, u0, bcs=None, basis_type=None,
+                 quadrature=None,
+                 solver_parameters=None, appctx=None, nullspace=None):
+        assert order >= 1
+        self.u0 = u0
+        self.orig_bcs = bcs
+        self.t = t
+        self.dt = dt
+        self.order = order
+        if basis_type is None:
+            basis_type = "Lagrange"
+        self.basis_type = basis_type
+
+        V = u0.function_space()
+        self.num_fields = len(V)
+
+        ufc_line = ufc_simplex(1)
+        if basis_type == "Lagrange":
+            self.trial_el = Lagrange(ufc_line, order)
+            self.test_el = DiscontinuousLagrange(ufc_line, order-1)
+        elif basis_type == "Bernstein":
+            self.trial_el = Bernstein(ufc_line, order)
+            if order == 1:
+                self.test_el = DiscontinuousLagrange(ufc_line, 0)
+            else:
+                self.test_el = DiscontinuousElement(
+                    Bernstein(ufc_line, order-1))
+        else:
+            raise NotImplementedError("Not implemented basis type")
```

Comment:
```suggestion
            self.trial_el = Lagrange(ufc_line, order, variant=basis_type)
            self.test_el = DiscontinuousLagrange(ufc_line, order-1, variant=basis_type)
```

Reply:
Similarly, we could add another `basis_type="integral"` with `IntegratedLegendre` and `Legendre`

Reply:
I'm not sure what the use for other Lagrange elements would be, could it possibly diagonalize the mass matrix?  We're in a Petrov-Galerkin setting, this would have to be worked out.

Reply:
If we want to support other kinds of Lagrange basis, maybe `basis_type` could be `"Lagrange"` or `"Lagrange,..."` where we put in a variant after the comma and split it off?

Reply:
I like how this is resolved now